services/api-server/tests/unit/with_db/conftest.py

```diff
@@ -0,0 +1,230 @@
+# pylint: disable=no-name-in-module
+# pylint: disable=protected-access
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+
+import os
+import shutil
+import subprocess
+from pathlib import Path
+from pprint import pformat
+from typing import Callable, Dict, Union
+
+import aiopg.sa
+import aiopg.sa.engine as aiopg_sa_engine
+import faker
+import passlib.hash
+import pytest
+import simcore_postgres_database.cli as pg_cli
+import simcore_service_api_server.db.tables as orm
+import sqlalchemy as sa
+import sqlalchemy.engine as sa_engine
+import yaml
+from aiopg.sa.result import RowProxy
+from faker import Faker
+from fastapi import FastAPI
+from pytest_simcore.helpers.typing_env import EnvVarsDict
+from simcore_postgres_database.models.base import metadata
+from simcore_service_api_server.core.application import init_app
+from simcore_service_api_server.db.repositories import BaseRepository
+from simcore_service_api_server.db.repositories.users import UsersRepository
+from simcore_service_api_server.models.domain.api_keys import ApiKeyInDB
+
+## POSTGRES -----
+
+
+@pytest.fixture(scope="session")
+def docker_compose_file(
+    default_app_env_vars: dict[str, str], tests_utils_dir: Path, tmpdir_factory
+) -> Path:
+    # Overrides fixture in https://github.com/avast/pytest-docker
+
+    # NOTE: do not forget to add the current environ here, otherwise docker-compose fails
+    environ = dict(os.environ)
+    environ.update(default_app_env_vars)
+
+    src_path = tests_utils_dir / "docker-compose.yml"
+    assert src_path.exists
+
+    dst_path = Path(str(tmpdir_factory.mktemp("config").join("docker-compose.yml")))
+
+    shutil.copy(src_path, dst_path.parent)
+    assert dst_path.exists()
+
+    # configs
+    subprocess.run(
+        f'docker-compose --file "{src_path}" config > "{dst_path}"',
+        shell=True,
+        check=True,
+        env=environ,
+    )
+
+    return dst_path
+
+
+@pytest.fixture(scope="session")
+def postgres_service(docker_services, docker_ip, docker_compose_file: Path) -> Dict:
```

Comment:
Thought: I see you are re-creating new postgres based on docker compose instead of docker swarm. I find this confusing. I thought you were using something in memory. Actually why not improve the starting time of the docker swarm postgres, that exactly fakes the real one instead of creating a docker-compose one? Especially since that means it's duplicated, and might be acting differently. In brief, I find that a bit dangerous. Since there are some differences between swarm and compose.

Reply:
hi @sanderegg. The docker-compose fixture was already there from previous versions. I just relocated it.
Nonetheless, I understand your point and will upgrade to docker-swarm